### PR TITLE
Add test for CRD server-dry-run and fix bug

### PR DESCRIPTION
**Pull request by apelisse**
Add a new test to make sure we can server-dry-run CRDs and also fix a
typo now that we have a test, we could notice that it doesn't work.

**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
Fixes a bug in `kubectl diff`

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Review by seans3 — APPROVED** [review]

File: pkg/kubectl/cmd/util/crdfinder.go [hunk 1]
```diff
@@ -34,10 +35,10 @@ func CRDFromDynamic(client dynamic.Interface) CRDGetter {
 		list, err := client.Resource(schema.GroupVersionResource{
 			Group:    "apiextensions.k8s.io",
 			Version:  "v1beta1",
-			Resource: "curstomresourcedefinitions",
```
Comment (yue9944882) [inline pkg/kubectl/cmd/util/crdfinder.go]: OMG😱

**Pull request merged**